include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "FacebookMessenger" nettest.
+class FacebookMessengerSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "FacebookMessenger";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "HttpHeaderFieldManipulation" nettest.
+class HttpHeaderFieldManipulationSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "HttpHeaderFieldManipulation";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "HttpInvalidRequestLine" nettest.
+class HttpInvalidRequestLineSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "HttpInvalidRequestLine";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "MeekFrontedRequests" nettest.
+class MeekFrontedRequestsSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "MeekFrontedRequests";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "MultiNdt" nettest.
+class MultiNdtSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "MultiNdt";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Ndt" nettest.
+class NdtSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Ndt";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "TcpConnect" nettest.
+class TcpConnectSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "TcpConnect";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Telegram" nettest.
+class TelegramSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Telegram";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "WebConnectivity" nettest.
+class WebConnectivitySettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "WebConnectivity";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Whatsapp" nettest.
+class WhatsappSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Whatsapp";
+    
+    /// The "all_endpoints" setting of this nettest.
+    bool all_endpoints = false;
+};
+
+} // namespace settings
+
+// Contains the routers
+namespace routers {
+
+/// Routes nettest events to virtual methods.
+class Router {
+  public:
+    // Implementation note: virtual methods cannot be `noexcept` because the
+    // task stable version of SWIG does not handle that correctly.
```

Comment:
(This seems to be fixed in master but does not work for me locally. I can probably investigate more.)